### DISCUSSION TITLE
Does bookwyrm have an API?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [bookwyrm-social/bookwyrm#2662] [Q&A] [answered]
I'd like to request that it be added as a tracker to Tachiyomi, but I don't know whether ActivityPub is enough to facilitate this. Bookwyrm at least supports OAuth, right?

**Comment by jaschaurbach** [comment]
There are several issues asking for an API. 

The questions which need to be discussed first: Scope of the API, should it be Mastodon compatible, if yes, how can this be achieved and some more.

OAUTH is under development (see #2659)

**Reply by roke-julian-lockhart** [reply]
> https://github.com/bookwyrm-social/bookwyrm/discussions/2662#discussioncomment-5176344

@jaschaurbach, so it doesn't, currently?

**Reply by mouse-reeve** [reply]
correct, we currently do not have much in the way of API (ie, anything outside of what activitypub uses)

**Comment by steevithak** [comment]
I'm hoping to migrate away from  Goodreads but need an API because I use it to display my activity on my Wordpress blog (for example). A simple REST API that could retrieve my reading activity and book cover images would be a good start. Or even just an activity feed in RSS or ATOM format could be used if something like that exists. Is there a specific ticket covering the need for an API that I could follow?

**Reply by andypiper** [reply]
There is an RSS feed ... at least for my user, on the instance I use -> bookrastinating.com/user/andypiper/rss

![image of the RSS feed link in the Bookwyrm user interface for a user profile page](https://github.com/user-attachments/assets/1463537a-d8c4-4be9-96f5-476eb6013458)

That could be a useful start for something like this.

It might be cool to have something more fully-featured, but I have no idea what that might look like.

**Comment by mateitudor** [comment]
Hi folks, jumping in with a small, concrete use-case that might help.
I’ve been experimenting with tools that manage local ebook libraries (think Calibre-like, but headless). Right now the only way they can see my “to-read” shelf is by scraping the HTML, which is fragile and breaks every time the markup changes. What would make life *so* much easier is a tiny, read-only endpoint:

- **Route** — `/user/<username>/shelf/<shelf>.json`  
- **No auth required** — public shelves are already, well, public.  
- **Lightweight JSON** — just `title`, `author`, `isbn_13`, maybe a couple of cover sizes and a stable `remote_id`.  
- **Paginated** — so I don’t slam the instance on the first sync.  
- **Caching headers** — so polite clients don’t poll more than once every few minutes.

That’s literally all I need to keep my offline library in sync with what I’m planning to read on Bookwyrm. No write access, no Mastodon compatibility layer, no OAuth dance, just a tiny, predictable JSON feed.

If others have similar “I only need a handful of fields” stories, maybe we can scope a *minimal* public API and grow it later instead of trying to design the full kitchen sink on day one.

Thanks for reading, and thanks for all the amazing work on Bookwyrm!

**Reply by mouse-reeve** [reply]
That actually does exist, and it’s at the route you suggest already

**Reply by roke-julian-lockhart** [reply]
@mouse-reeve, because it's what AP provides? [^1]

[^1]: [`#discussioncomment-7621484`][1]

[1]: https://github.com/bookwyrm-social/bookwyrm/discussions/2662#discussioncomment-7621484

**Reply by mouse-reeve** [reply, marked as answer]
Public data is fairly accessible through the ActivityPub API, the need is really for an authenticated API

**Reply by mateitudor** [reply]
Ooh, understood, sorry, I am a newbie. This makes sense.
